src/legacy/core_plugins/console/public/np_ready/application/models/legacy_core_editor/create_readonly.ts

```diff
@@ -18,19 +18,28 @@
  */
 
 import _ from 'lodash';
-const ace = require('brace');
-const OutputMode = require('./sense_editor/mode/output');
-const smartResize = require('./smart_resize');
+import ace from 'brace';
+// @ts-ignore
+import * as OutputMode from './mode/output';
+import smartResize from './smart_resize';
 
-let output;
-export function initializeOutput($el) {
-  output = ace.acequire('ace/ace').edit($el[0]);
+export interface CustomAceEditor extends ace.Editor {
+  update: (text: string, mode?: any, cb?: () => void) => void;
+  append: (text: string, foldPrevious?: boolean, cb?: () => void) => void;
+}
+
+/**
+ * Note: using read-only ace editor leaks the Ace editor API - use this as sparingly as possible or
+ * create an interface for it so that we don't rely directly on vendor APIs.
+ */
+export function createReadOnlyAceEditor(element: HTMLElement): CustomAceEditor {
+  const output = ace.acequire('ace/ace').edit(element);
 
   const outputMode = new OutputMode.Mode();
 
   output.$blockScrolling = Infinity;
   output.resize = smartResize(output);
-  output.update = (val, mode, cb) => {
+  output.update = (val: string, mode?: any, cb?: any) => {
```

Comment:
This definition (and `append` below) don't agree with the `CustomAceEditor` interface defined above. Is that intentional?

Reply:
Nope - will fix :+1: